ironfish/src/network/peerNetwork.ts

```diff
@@ -1197,6 +1197,11 @@ export class PeerNetwork {
       return
     }
 
+    if (block.header.sequence === GENESIS_BLOCK_SEQUENCE) {
+      peer.punish(BAN_SCORE.MAX, 'Peer gossiped its genesis block')
+      return
```

Comment:
We need to do this for compact blocks as well -- We won't ever hit this code without receiving a compact block first.

Also, this should be done next to the block header verification call since it's also verifying the block header, and should call `this.blockFetcher.removeBlock(block.header.hash)` (and probably `this.chain.addInvalid`)

Reply:
Added this check for compact blocks as well and moved it next to the block header verification call (with the additional `removeBlock and `addInvalid` calls). Also tested this locally.